src/librbd/AioRequest.cc

```diff
@@ -135,36 +143,46 @@ namespace librbd {
 
       // This is the step to read from parent
       if (!m_tried_parent && r == -ENOENT) {
-        RWLock::RLocker l(m_ictx->snap_lock);
-        RWLock::RLocker l2(m_ictx->parent_lock);
-
-        if (m_ictx->parent == NULL) {
-	  ldout(m_ictx->cct, 20) << "parent is gone; do nothing" << dendl;
-	  m_state = LIBRBD_AIO_READ_FLAT;
-	  finished = false;
-	  break;
-	}
+        {
+          RWLock::RLocker l(m_ictx->snap_lock);
+          RWLock::RLocker l2(m_ictx->parent_lock);
+          if (m_ictx->parent == NULL) {
+	    ldout(m_ictx->cct, 20) << "parent is gone; do nothing" << dendl;
+	    m_state = LIBRBD_AIO_READ_FLAT;
+	    finished = false;
```

Comment:
hi @dillaman , i am wondering why we have to set `finished` to `false` here (same on the master https://github.com/ceph/ceph/blob/master/src/librbd/io/ObjectRequest.cc#L207)?
since the parent object has gone, and the child object does not exist, who will drive us to finish the original read request?
am i wrong? thank you very much.